src/mqt/ddsim/sampler.py

```diff
@@ -0,0 +1,144 @@
+"""Sampler implementation for an artibtrary Backend object."""
+
+from __future__ import annotations
+
+import math
+from typing import TYPE_CHECKING, Any, Mapping, Sequence, Union
+
+# from qiskit.primitives.backend_estimator import _prepare_counts
+from qiskit.primitives.base import BaseSampler, SamplerResult
+from qiskit.primitives.primitive_job import PrimitiveJob
+from qiskit.providers.options import Options
+from qiskit.result import QuasiDistribution, Result
+
+from mqt.ddsim.qasmsimulator import QasmSimulatorBackend
+
+if TYPE_CHECKING:
+    from qiskit.circuit import Parameter
+    from qiskit.circuit.parameterexpression import ParameterValueType
+    from qiskit.circuit.quantumcircuit import QuantumCircuit
+    from qiskit.transpiler.passmanager import PassManager
+
+    Parameters = Union[Mapping[Parameter, ParameterValueType], Sequence[ParameterValueType]]
+
+
+def prepare_counts(results: list[Result]):
+    counts = []
+    for res in results:
+        count = res.get_counts()
+        if not isinstance(count, list):
+            count = [count]
+        counts.extend(count)
+    return counts
+
+
+class DDSIMBackendSampler(BaseSampler):
```

Comment:
```suggestion
class Sampler(BaseSampler):
```
See comment in the notebook